tools/autograd/templates/ViewFuncs.cpp

```diff
@@ -0,0 +1,69 @@
+#include "torch/csrc/autograd/generated/ViewFuncs.h"
+
+// ${generated_comment}
+
+using at::Tensor;
+using at::Scalar;
+using at::IntArrayRef;
+using at::TensorList;
+
+namespace torch::autograd {
+
+ChainedViewFunc::ChainedViewFunc(
+  const std::shared_ptr<ViewFunc>& first_,
+  const std::shared_ptr<ViewFunc>& second_)
+  : first(first_),
+    num_first_symints(-1),
+    num_first_tensors(-1),
+    second(second_),
+    num_second_symints(-1),
+    num_second_tensors(-1) {}
+
+std::vector<c10::SymInt> ChainedViewFunc::get_symints() {
+  auto symints = first->get_symints();
+  num_first_symints = symints.size();
+  auto second_symints = second->get_symints();
+  num_second_symints = second_symints.size();
+  symints.reserve(symints.size() + num_second_symints);
```

Comment:
I think it's still useful in certain cases, since AFAICT `insert()` won't calculate total size beforehand ([citation](https://stackoverflow.com/questions/35359169/does-stdvectorinsert-reserve-by-definition))